lib/Mojo/WebSocketProxy/Backend/JobAsync.pm

```diff
@@ -137,19 +157,35 @@ sub call_rpc {
             my $api_response;
 
             if ($f->is_done) {
-                my $result = MojoX::JSON::RPC::Client::ReturnObject->new(rpc_response => decode_json_utf8($f->get));
-
-                $_->($c, $req_storage, $result) for @$after_got_rpc_response_hook;
-
-                $api_response = $rpc_response_cb->($result->result);
-                stats_inc("rpc_queue.client.jobs.success", {tags => ["rpc:" . $req_storage->{name}, 'clientID:' . $self->client->id]});
+                try {
+                    my $result = MojoX::JSON::RPC::Client::ReturnObject->new(rpc_response => decode_json_utf8($f->get));
+
+                    $_->($c, $req_storage, $result) for @$after_got_rpc_response_hook;
```

Comment:
if there was an issue with the call and no result was returned this is not called in the non async version.

Reply:
Again running if future is done. There is no failure here.

Reply:
ok